torch/onnx/_internal/jit_utils.py

```diff
@@ -321,3 +321,13 @@ def get_device_from_value(value: _C.Value) -> Optional[torch.device]:
         return None
     tensor_type = typing.cast(_C.TensorType, value.type())
     return tensor_type.device()
+
+
+@_beartype.beartype
+def _parse_node_kind(kind: str) -> Tuple[str, str]:
+    if "::" not in kind:
+        raise torch.onnx.errors.OnnxExporterError(
+            "Node kind is invalid. '::' is not in node kind."
```

Comment:
Please also print kind in the error message.

Reply:
btw I'm curious, did you observe failure cases that "::" was not in 'kind'? And should we check if the return from `kind.split("::")` has precisely 2 elements? (or just `.split("::", 1)`)

Reply:
> btw I'm curious, did you observe failure cases that "::" was not in 'kind'?

For _C.node.kind() probably no, but this function is not only for the it. Anyone has string domain::opname should be able to use this function, so I think the bottom line should be there is a '::' needed in input.

> And should we check if the return from kind.split("::") has precisely 2 elements? (or just .split("::", 1))

I think if there is '::' in it. It will be a two elements list, even the input is '::add' (this case domain would be ''). But I think this is out of scope on this function. LMK what you think.

Reply:
Yea just checking for bad inputs, like "ns1::ns2::op"

Reply:
Ah, I see.

Reply:
Done